lib/ex_libsrt/server.ex

```diff
@@ -54,33 +63,55 @@ defmodule ExLibSRT.Server do
   Starts a new SRT server binding to given address and port and links to current process.
 
   One may usually want to bind to `0.0.0.0` address.
+
+  ## Password Requirements
+
+  If a password is provided, it must be between 10 and 79 characters long according to SRT specification.
+  An empty string means no password authentication will be used.
   """
   @spec start_link(address :: String.t(), port :: non_neg_integer()) ::
           {:ok, t()} | {:error, reason :: String.t(), error_code :: integer()}
-  def start_link(address, port) do
-    case ExLibSRT.Native.start_server(address, port) do
-      {:ok, server_ref} ->
-        Agent.start_link(fn -> server_ref end)
-
-      {:error, reason, error_code} ->
-        {:error, reason, error_code}
+  @spec start_link(address :: String.t(), port :: non_neg_integer(), password :: String.t()) ::
+          {:ok, t()} | {:error, reason :: String.t(), error_code :: integer()}
+  def start_link(address, port, password \\ "") do
+    with :ok <- validate_password(password) do
+      case ExLibSRT.Native.start_server(address, port, password) do
+        {:ok, server_ref} ->
+          Agent.start_link(fn -> server_ref end)
+
+        {:error, reason, error_code} ->
+          {:error, reason, error_code}
+      end
+    else
+      {:error, reason} -> {:error, reason, 0}
     end
```

Comment:
Could you rewrite it so that not to have this nested `case` clause? I think it would be OK to do it with two subsequent match clauses within `with` statement.
It also applies to the corresponding code in the `client` implementation